src/main/java/de/ustutt/iaas/cc/core/QueueTextProcessor.java

```diff
@@ -21,24 +42,83 @@
  */
 public class QueueTextProcessor implements ITextProcessor {
 
-	private final static Logger logger = LoggerFactory.getLogger(QueueTextProcessor.class);
+	final Logger logger = LoggerFactory.getLogger(QueueTextProcessor.class);
+
+	private QueueConnectionFactory conFactory;
+	private QueueConnection connection;
+	private QueueSession session;
+	private Queue textProcessorRequests;
+	private Queue textProcessorResponses;
+	private QueueSender textProcessorRequestSender;
+	private QueueReceiver textProcessorResponseReceiver;
+	private AsyncReceiver asyncReceiver;
+	
+	private ConcurrentHashMap<UUID, CompletableFuture<String>> responseMap ;

```

Comment:
I would declare this as a static map to ensure that there is in any case one global map for all responses. Currently, there exists only one instance of the TextProcessor component, so this is not a problem, but you never know what happens in the future.